### PR TITLE
Fixed the timestamp error 

### PR DESCRIPTION
**Pull request by sreejakshetty**
#78

### CODE REVIEW

**Comment by sreejakshetty** [comment]
screenshot from 2016-06-10 19 34 07
screenshot from 2016-06-10 19 33 50
screenshot from 2016-06-10 19 33 35
screenshot from 2016-06-10 19 33 09
screenshot from 2016-06-10 19 32 54

**Comment by sreejakshetty** [comment]
tested on online server

**Comment by DishantK1807** [comment]
@kamishettysreeja25 : I see here that you changed install.sql file here. Please make those changes in update.sql file as we are not changing the install.sql file.

**Comment by sreejakshetty** [comment]
@DishantK1807 sure. I didn't know about that. I will add the fields in update.sql and undo them in install.sql.
Thanks

**Comment by sreejakshetty** [comment]
I have undo the changes made in install.sql and added table entries in update.sql.Please review the PR

**Comment by hongquan** [comment]
@kamishettysreeja25 I recently merged some PR and make this PR have conflict. Can you please merge from master to this branch to fix the conflict?

**Comment by sreejakshetty** [comment]
@hongquan  I have fixed the conflicts . Can you review it and merge it. 
Thanks :)

File: includes/pages/user_shifts.php [hunk 1]
```diff
@@ -39,7 +42,7 @@ function user_shifts() {
       if ($result === false)
         engelsystem_error('Unable to delete shift entry.');
 
-      engelsystem_log("Deleted " . User_Nick_render($shift_entry_source) . "'s shift: " . $shift_entry_source['name'] . " at " . $shift_entry_source['Name'] . " from " . date("Y-m-d H:i", $shift_entry_source['start']) . " to " . date("Y-m-d H:i", $shift_entry_source['end']) . " as " . $shift_entry_source['angel_type']);
+      engelsystem_log("Deleted " . User_Nick_render($shift_entry_source) . "'shift: " . $shift_entry_source['name'] . " at " . $shift_entry_source['Name'] . " from " . date("Y-m-d", $shift_entry_source['start']) . " " . date("H:i", $shift_entry_source['start_time']) . " to " . date("Y-m-d", $shift_entry_source['end']) . " " . date("H:i", $shift_entry_source['end_time'])  . " as " . $shift_entry_source['angel_type']);
```
Comment (hongquan) [inline includes/pages/user_shifts.php]: Why is there change from "someone's shift" to "someone'shift"?

**Comment by sreejakshetty** [comment]
@hongquan 
I have mistakenly changed it . I will change it to " someone's shift ".
Thank you

**Comment by hongquan** [comment]
Please clean trailing whitespaces (I reminded this in Slack chat)

**Comment by sreejakshetty** [comment]
@hongquan I have indented the code and removed the trailing spaces . 
Thanks.

**Comment by hongquan** [comment]
@kamishettysreeja25 Please read again: https://www.drupal.org/coding-standards#functcall
- There should be no space around the brackets

```
$var = foo($bar, $baz, $quux);
```
- The indentation width is 2 spaces, not 8 spaces (https://www.drupal.org/coding-standards#indenting). The original code follows this well.
- There should be one blank line at the end of file (you can configure editor to add it automatically)

**Comment by sreejakshetty** [comment]
@hongquan Sure . I will follow this from now. I have pushed the code before I read the coding conventions given by you.I will make necessary changes and push the code.
Cheers

**Comment by sreejakshetty** [comment]
@hongquan I have changed the files . Added 2 space instead of 8 spaces and removed extra spaces at end of brackets " ) " and added one blank line at the end of file.

**Comment by hongquan** [comment]
Why do you change the braces from

``` php
function do() {
  // blah blah
}
```

to

``` php

function do()
{
  // blah blah
}
```

The original code is already compliant with Drupal convention (https://www.drupal.org/coding-standards#functdecl).

**Comment by sreejakshetty** [comment]
@hongquan I have used a php formatter and it created the indentation from 

```
fun(){ 
// code
}
```

to 

``` fun
fun(){
//code
}
```

now , I have changed it .
Thank you

**Comment by hongquan** [comment]
There still be a lot of trailing spaces. Please look into https://github.com/fossasia/engelsystem/pull/87/files to check.

There are some blank lines to separate block of codes, to make it easier to read. Don't delete them (but still clean the trailing spaces on them).

Some lines have wrong indentation.

File: includes/pages/admin_shifts.php [hunk 2]
```diff
@@ -195,22 +191,22 @@ function admin_shifts() {
...
             $shift_end = $end ;
           if ($shift_start >= $shift_end)
             break;
           $shifts[] = array(
               'start' => $shift_start,
               'end' => $shift_end,
-              'start_time' => $start_time,
-              'end_time'  => $end_time,
+              'start_time' => DateTime::createFromFormat("H:i", date("H:i",$shift_start))->getTimestamp(),
+              'end_time'  =>DateTime::createFromFormat("H:i", date("H:i",$shift_end))->getTimestamp(),
```
Comment (hongquan) [inline includes/pages/admin_shifts.php]: You lack space after `=>`

File: includes/pages/admin_shifts.php [hunk 3]
```diff
@@ -195,22 +191,22 @@ function admin_shifts() {
...
               'RID' => $rid,
               'title' => $title,
               'shifttype_id' => $shifttype_id 
           );
           
           $shift_start = $shift_end;
-        } while ($shift_end < $end);
+        } while ($shift_end < $end );
```
Comment (hongquan) [inline includes/pages/admin_shifts.php]: Why the space before `)`?

File: includes/pages/user_settings.php [hunk 4]
```diff
@@ -5,7 +5,7 @@ function settings_title() {
 }
 
 function user_settings() {
-  global $enable_tshirt_size, $tshirt_sizes, $themes, $locales;
+  global $enable_tshirt_size, $tshirt_sizes, $themes, $locales ;
```
Comment (hongquan) [inline includes/pages/user_settings.php]: why space between `$locales` and `;`?

File: includes/pages/user_settings.php [hunk 5]
```diff
@@ -200,9 +206,61 @@ function user_settings() {
...
                   form_select('language', _("Language:"), $locales, $selected_language),
                   form_submit('submit_language', _("Save")) 
+              ))
+          ))
+      )) 
+  ));
+}
+
+// User settings page
+if( $_SESSION['uid'] > 1){
```
Comment (hongquan) [inline includes/pages/user_settings.php]: Please fix the space

File: includes/pages/user_shifts.php [hunk 6]
```diff
@@ -4,6 +4,10 @@ function shifts_title() {
   return _("Shifts");
 }
 
+function edit_shifts_title(){
```
Comment (hongquan) [inline includes/pages/user_shifts.php]: Please fix whitespace

File: includes/pages/user_shifts.php [hunk 7]
```diff
@@ -294,7 +316,7 @@ function user_shifts() {
         sql_query("INSERT INTO `UserAngelTypes` (`user_id`, `angeltype_id`) VALUES ('" . sql_escape($user_id) . "', '" . sql_escape($selected_type_id) . "')");
 
       $user_source = User($user_id);
```
Comment (hongquan) [inline includes/pages/user_shifts.php:330]: Why there is difference in indentation between the line

```
sql_query("INSERT INT
```

and

```
$user_source = 
```

**Comment by sreejakshetty** [comment]
#87 #99 #91 all these issues are fixed
Tested on online server http://107.170.88.225/engelsystem/public/
screenshot from 2016-06-15 07 47 07

**Comment by sreejakshetty** [comment]
@hongquan Removed all trailing spaces as mentioned . 
Thank you.

**Comment by hongquan** [comment]
@kamishettysreeja25 Don't fix only on the line I commented. You should be proactive. There are too many issues that I cannot comment all. The issue which I commented is repeated in other parts of file and other files in your commits. I told you to look into https://github.com/fossasia/engelsystem/pull/87/files to check on all your files.

**Comment by sreejakshetty** [comment]
@hongquan This PR even contains files which are already pushed into the master like user_settings.php and guest_login.php . The files I changed are admin_shifts.php , user_shifts.php , Shifts_model.php,update.sql.I would like to  create a new PR as this PR contains the merge commits and others files also .
Thank you.

**Comment by sreejakshetty** [comment]
Fixed #106 here

**Comment by hongquan** [comment]
Did you **really** go to the link? The link only list the changes made by you. I only tell you to check the change **made by you**. Those already in master are even not shown there. Don't blame the master.

selection_094

**Comment by sreejakshetty** [comment]
@hongquan I was talking about the commits which were merged into master and which were present in this branch. I am not blaming master .
screenshot from 2016-06-15 11 23 16
I have gone through the link went through the code and I have made necessary changes to the files and create a new PR #106 , as this PR contains the files which were already merged and has merge  commits.
screenshot from 2016-06-15 11 17 13
screenshot from 2016-06-15 11 16 25
Cheers and Thank you :)

**Comment by hongquan** [comment]
What is the purpose of #106? It is to replace this PR and this PR should be closed (without merging)?
- If yes, why do you have to create a new PR just to do the same thing as this PR?
- If no, what is the difference between this PR and #106?

The links https://github.com/fossasia/engelsystem/pull/87/files show the difference between your branch and master. That difference is caused by your change of code in your branch. Any files which is in master and is not different between your branch and master will not be shown in that links.

Even if you merge _from master to your branch_, brings some erroneous files to your branch, but you do not do anything, those error parts won't be shown here, because the difference is zero.

The change that I showed you in the picture above is the change made by this PR. That change (and other similar) is not qualified and this PR cannot be merged. If you refuse to fix, this PR can only be closed.

**Comment by hongquan** [comment]
> this PR contains the files which were already merged

This argument is irrelevant, because the difference on the file, between your branch and master will be zero and it won't be shown here for me to review.

**Comment by sreejakshetty** [comment]
> What is the purpose of #106? It is to replace this PR and this PR should be closed (without merging)?
> -    If yes, why do you have to create a new PR just to do the same thing as this PR?
> -    If no, what is the difference between this PR and #106?

#106 takes care of both PR #87  #83  .These both PR's can be closed.
screenshot from 2016-06-15 13 20 16
As I see in the link https://github.com/fossasia/engelsystem/pull/87/files . I see 8 files being changed .
I taught there is a conflict in my branch as I fetched the code from master and pushed it to error_shifts branch.
So I created a new PR to tackle this issue. I have fixed all the issues in #106 .

**Pull request closed without merging**